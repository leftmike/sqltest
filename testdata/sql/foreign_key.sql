--
-- Test FOREIGN KEY constraints
--
-- {{if eq Dialect "sqlite3"}}{{Skip}}{{end}}

DROP TABLE IF EXISTS tbl2;

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (
    c1 int,
    c2 int PRIMARY KEY,
    c3 int
);

CREATE TABLE tbl2 (
    c4 int PRIMARY KEY,
    c5 int,
    c6 int REFERENCES tbl1
);

INSERT INTO tbl1 VALUES
    (10, 1, 100),
    (20, 2, 200),
    (30, 3, 300),
    (40, 4, 400),
    (50, 5, 500);

INSERT INTO tbl2 VALUES
    (10, 100, 1),
    (20, 200, 2),
    (30, 300, 3);

{{Fail .Test}}
INSERT INTO tbl2 VALUES
    (90, 900, 9);

{{Fail .Test}}
INSERT INTO tbl2 VALUES
    (40, 400, 4),
    (50, 500, 5),
    (60, 600, 6);

SELECT * FROM tbl1;

SELECT * FROM tbl2;

INSERT INTO tbl1 VALUES
    (60, 6, 600);

INSERT INTO tbl2 VALUES
    (40, 400, 4),
    (50, 500, 5),
    (60, 600, 6),
    (70, 700, 1),
    (80, 800, 2),
    (90, 900, 3);

SELECT * FROM tbl1;

SELECT * FROM tbl2;

-- Drop table with foreign key reference first.
DROP TABLE IF EXISTS tbl2;

DROP TABLE IF EXISTS tbl1;
