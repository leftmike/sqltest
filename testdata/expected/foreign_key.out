--
-- Test FOREIGN KEY constraints
--
-- {{if eq Dialect "sqlite3"}}{{Skip}}{{end}}
DROP TABLE IF EXISTS tbl2;
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (
    c1 int,
    c2 int PRIMARY KEY,
    c3 int
);
CREATE TABLE tbl2 (
    c4 int PRIMARY KEY,
    c5 int,
    c6 int REFERENCES tbl1 ON DELETE RESTRICT ON UPDATE RESTRICT
);
INSERT INTO tbl1 VALUES
    (10, 1, 100),
    (20, 2, 200),
    (30, 3, 300),
    (40, 4, 400),
    (50, 5, 500);
INSERT INTO tbl2 VALUES
    (10, 100, 1),
    (20, 200, 2),
    (30, 300, 3);
{{Fail .Test}}
INSERT INTO tbl2 VALUES
    (90, 900, 9);
{{Fail .Test}}
INSERT INTO tbl2 VALUES
    (40, 400, 4),
    (50, 500, 5),
    (60, 600, 6);
SELECT * FROM tbl1;
   c1 c2  c3
   -- --  --
 1 10  1 100
 2 20  2 200
 3 30  3 300
 4 40  4 400
 5 50  5 500
(5 rows)
SELECT * FROM tbl2;
   c4  c5 c6
   --  -- --
 1 10 100  1
 2 20 200  2
 3 30 300  3
(3 rows)
INSERT INTO tbl1 VALUES
    (60, 6, 600);
INSERT INTO tbl2 VALUES
    (40, 400, 4),
    (50, 500, 5),
    (60, 600, 6),
    (70, 700, 1),
    (80, 800, 2),
    (90, 900, 3);
SELECT * FROM tbl1;
   c1 c2  c3
   -- --  --
 1 10  1 100
 2 20  2 200
 3 30  3 300
 4 40  4 400
 5 50  5 500
 6 60  6 600
(6 rows)
SELECT * FROM tbl2;
   c4  c5 c6
   --  -- --
 1 10 100  1
 2 20 200  2
 3 30 300  3
 4 40 400  4
 5 50 500  5
 6 60 600  6
 7 70 700  1
 8 80 800  2
 9 90 900  3
(9 rows)
{{Fail .Test}}
DELETE FROM tbl1 WHERE c2 = 6;
{{Fail .Test}}
UPDATE tbl1 SET c2 = 44 WHERE c2 = 4;
INSERT INTO tbl1 VALUES
    (70, 7, 700),
    (80, 8, 800),
    (90, 9, 900);
SELECT * FROM tbl1;
   c1 c2  c3
   -- --  --
 1 10  1 100
 2 20  2 200
 3 30  3 300
 4 40  4 400
 5 50  5 500
 6 60  6 600
 7 70  7 700
 8 80  8 800
 9 90  9 900
(9 rows)
{{Fail .Test}}
DELETE FROM tbl1 WHERE c2 > 3;
{{Fail .Test}}
UPDATE tbl1 SET c2 = c2 * 10 WHERE c2 > 5;
SELECT * FROM tbl1;
   c1 c2  c3
   -- --  --
 1 10  1 100
 2 20  2 200
 3 30  3 300
 4 40  4 400
 5 50  5 500
 6 60  6 600
 7 70  7 700
 8 80  8 800
 9 90  9 900
(9 rows)
-- Drop table with foreign key reference first.
DROP TABLE IF EXISTS tbl2;
DROP TABLE IF EXISTS tbl1;
